core/src/main/java/hudson/model/Queue.java

```diff
@@ -3051,8 +3051,9 @@ public void onLeft(Queue.LeftItem li) {
         private void push() {
             if (DELAY_SECONDS < 0) return;
 
+            Future<?> ns = nextSave; // Atomic read
             // Can be done or canceled in case of a bug or external intervention - do not allow it to hang there forever
-            if (nextSave != null && !(nextSave.isDone() || nextSave.isCancelled())) return;
+            if (ns != null && !(ns.isDone() || ns.isCancelled())) return;
             nextSave = Timer.get().schedule(this, DELAY_SECONDS, TimeUnit.SECONDS);
```

Comment:
Still thread-unsafe due to a race condition: a second thread can fetch the field, see it's empty, and schedule a *second* task while the first is scheduling a Timer task but before it writes back to the field. 

Just synchronize the darn thing.  That's what synchronization is there for.  `Volatile` should really be reserved for `booleans`, `int`s, etc where it can be used more safely -- and even then 90% of the time you need the `AtomicX` compare & set / increment operations.